public/app/core/components/Select/FolderPicker.tsx

```diff
@@ -43,6 +43,8 @@ export interface Props {
   accessControlMetadata?: boolean;
   customAdd?: CustomAdd;
   folderWarning?: FolderWarning;
+  sliceResults?: number;
+  virtualizeResults?: boolean;
```

Comment:
Any specific reason we allow consumers to change this?

I don't think these should be props. I think FolderPicker should be a very specific and opinionated component with a consistent as possible experience everywhere.

Reply:
yes, my idea was applying this change only to `RuleFolderPicker`, and once we see that it works as we expect, remove the prop and extend the functionality to the rest of consumers. Do you think it's better to apply the changes directly to the FolderPicker?

Reply:
removing props is a breaking change, so if it's our intention for these to be temporary then I think they should be named as such to discourage others from using/relying on them.

I don't think this is particuarly risky - do you?. I'll discuss today with the rest of the team, but I think we should be confident enough to just use this everywhere.

Besides, we've had a bunch of regressions with FolderPicker and search, so I would want to test thoroughly either way :)

Reply:
> removing props is a breaking change, so if it's our intention for these to be temporary then I think they should be named as such to discourage others from using/relying on them.
💯 you're right!

> I don't think this is particuarly risky - do you?.
I don't, honestly, I was not sure about how to proceed, but I agree with you, we should apply the changes for al consumers.

Reply:
> I'll discuss today with the rest of the team

thank you @joshhunt !

Reply:
I'm thinking that maybe it's a good idea, letting consumers decide the value for the limit? So having a default limit but applying `sliceResults` prop in case of having it? wdyt @joshhunt

Reply:
We discussed this today - we agree that FolderPicker should have a consistent experience everywhere, and should not expose these props or controls to users. I cannot think of a reason why consumers would want to pick a different size for the 'slicing' - if the default for FolderPicker is not suitable for a particular use case, it should be not suitable for everyone.

Reply:
it make sense, I will update the PR then 👍

Reply:
I've just updated the PR making always the slice in FolderPicker ✅ @joshhunt 👀